airflow/models/dag.py

```diff
@@ -2948,7 +2948,7 @@ def bulk_write_to_db(
             else:
                 orm_dag.calculate_dagrun_date_fields(dag, last_automated_data_interval)
 
-            dag_tags = set(dag.tags or {})
+            dag_tags = dag.tags or set()
```

Comment:
Is this line still needed?

Reply:
We don't need it as we enforce it at the `c'tor`, so we can just use it directly.